packages/form-js-carbon-styles/src/carbon-styles.js

```diff
@@ -199,6 +197,108 @@ const ANCHOR_STYLES = css`
   }
 `;
 
+const READONLY_STYLES = css`
+  ${({ theme }) => css`
+    .fjs-container {
+      .fjs-readonly {
+
+        .fjs-input:read-only:not(:disabled), 
+        .fjs-textarea:read-only:not(:disabled),
+        .fjs-select:read-only:not(:disabled),
+        &.fjs-taglist,
+        .fjs-input-group {
+          background-color: transparent;
+        }
+
+        &.fjs-form-field-number {
+          .fjs-input-group .fjs-number-arrow-container {
+            background-color: transparent;
+
+            .fjs-number-arrow-up, 
+            .fjs-number-arrow-down {
+              background-color: transparent;
+              pointer-events: none;
+            }
+          }
+        }
+
+        &.fjs-form-field:not(.fjs-form-field-datetime) {
+          .fjs-input-group .fjs-input-adornment {
+            background: transparent;
+          }
+        }
+
+        &.fjs-form-field-select {
+          .fjs-input-group {
+            cursor: unset;
+            background-image: ${getSelectArrowSvg(theme.iconDisabled)};
+            background-color: transparent;
+            border-bottom: 1px solid var(--cds-border-strong, var(--cds-border-strong-01));
```

Comment:
I think you don't need this fallback, it's a Carbon internal thing and it should be covered already if you set up the Carbon properly.

Did you find any issues with it?

```suggestion
            border-bottom: 1px solid var(--cds-border-strong);
```